editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/preferences/ParameterValidator.java

```diff
@@ -0,0 +1,34 @@
+/******************************************************************************* 
+ * Copyright (c) 2017 Red Hat, Inc. 
+ * Distributed under license by Red Hat, Inc. All rights reserved. 
+ * This program is made available under the terms of the 
+ * Eclipse Public License v1.0 which accompanies this distribution, 
+ * and is available at http://www.eclipse.org/legal/epl-v10.html 
+ * 
+ * Contributors: 
+ * Red Hat, Inc. - initial API and implementation 
+ ******************************************************************************/
+package org.fusesource.ide.camel.editor.preferences;
+
+import org.eclipse.jface.dialogs.IInputValidator;
+import org.fusesource.ide.camel.editor.internal.UIMessages;
+
+/**
+ * 
+ * @author Andrej Podhradsky (apodhrad@redhat.com)
+ *
+ */
+public class ParameterValidator implements IInputValidator {
+
+	@Override
+	public String isValid(String newText) {
+		if (newText.isEmpty()) {
+			return UIMessages.preferredLabelsErrorMessageEmptyParameter;
+		}
+		if (newText.matches(".*[\\W&&[^-]].*")) {
```

Comment:
it is the same than org.fusesource.ide.camel.editor.preferences.ComponentValidator.isValid(String), what about factorizing it?

Reply:
these two regular expressions are different but I will refactor it so that we can use constants from that class.